content/blog/terragrunt-envs-management.md

```diff
@@ -0,0 +1,162 @@
+variables to pass to the module.
+
+```hcl
+# terragrunt.hcl
+terraform {
+  source = "gittfr:///exampleorg/app/aws?version=5.8.1"
+}
+inputs = {
+  app_name = "my-app"
+  environment = "production"
+}
+```
+
+Similarly in the other terragrunt files we define the module to use and the variables to pass to the module.
+and now for the storing the state file we can use the remote state file.
+
+```hcl
+# terragrunt.hcl
+generate "backend" {
+  path      = "backend.tf"
+  if_exists = "overwrite_terragrunt"
+  contents = <<EOF
+terraform {
+  backend "s3" {
+    bucket         = "my-terraform-state"
+    key            = "${path_relative_to_include()}/terraform.tfstate"
+    region         = "us-east-1"
+    encrypt        = true
+    dynamodb_table = "my-lock-table"
+  }
+}
+EOF
+}
+```
+
+This tells terragrunt to create a new file `backend.tf` with the contents of the `contents` block, Even if backend.tf already exists it will overwrite the contents because we have specified `overwrite_terragrunt`.
+This is like a defining a function, Now we need to call it in the files i.e `terragrunt.hcl` files.
+
+> This would look like this
+
+```hcl
+include "root" {
+  path = find_in_parent_folders()
+}
+```
+
+here the `find_in_parent_folders` returns the absolute path to the first `terragrunt.hcl` file it finds in the parent folders above the current `terragrunt.hcl` file.
+and the include block tells terragrunt to include all the cofigurations from the file with the path returned by the `find_in_parent_folders` function.
+
+## The Terragrunt Workflow
+
+Here is a high-level overview of the terragrunt workflow:
+
+> We run the command `terragrunt run-all plan` in the root directory.
+
+This will run the terragrunt plan in each sub directory it finds, Then it reads the `terragrunt.hcl` file in the current directory and does the following:
+
+1. Terragrunt reads the `terraform` block and downloads the specified Terraform module
+2. Terragrunt reads the `inputs` block and passes the variables to the Terraform module
+3. Terragrunt reads the `generate` block and creates the `backend.tf` file
+4. Terragrunt runs Terraform with the specified configurations
+5. Terragrunt stores the Terraform state in the remote state file
+6. Terragrunt returns the output of the Terraform run
+
+This workflow allows me to manage multiple environments with ease. I can make changes to my infrastructure code in a single place and apply those changes across all environments with a single command. Terragrunt’s ability to keep my Terraform code DRY has saved me time and effort, and I no longer have to worry about managing multiple sets of nearly identical infrastructure code.
+
+## How Terragrunt worked for me
+
+1. Centralized Modules: I now have a shared repository of modules that are environment-agnostic. The modules are reusable and cover things like VPCs, ECS clusters, RDS instances, and IAM roles. I no longer need to maintain three different sets of Terraform code.
```

Comment:
can you bold the words before `:` ?

Reply:
Addressed in [b3e3636](https://github.com/infraspecdev/infraspec.dev/pull/58/commits/b3e363637fcadd0a70471151bf3e25c6b0b35521)